clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -730,6 +730,15 @@ void CodeGenFunction::EmitOpenCLKernelMetadata(const FunctionDecl *FD,
         llvm::ConstantAsMetadata::get(Builder.getInt32(1))};
     Fn->setMetadata("stall_enable", llvm::MDNode::get(Context, AttrMDArgs));
   }
+
+  if (const SYCLIntelFPGAMaxConcurrencyAttr *A =
+          FD->getAttr<SYCLIntelFPGAMaxConcurrencyAttr>()) {
+    const auto *CE = dyn_cast<ConstantExpr>(A->getNThreadsExpr());
+    Optional<llvm::APSInt> ArgVal = CE->getResultAsAPSInt();
```

Comment:
```suggestion
    llvm::APSInt ArgVal = CE->getResultAsAPSInt();
```
The function doesn't return an `Optional`, so I assume this was unintentional.